code/controllers/subsystems/records.dm

```diff
@@ -0,0 +1,137 @@
+/var/datum/controller/subsystem/records/SSrecords
+
+/datum/controller/subsystem/records
+    name = "Records"
+    flags = SS_NO_FIRE
+
+    var/list/records
+
+/datum/controller/subsystem/records/New()
+    LAZYINITLIST(records)
+    NEW_SS_GLOBAL(SSrecords)
+
+/datum/controller/subsystem/records/proc/generate_record(var/mob/living/carbon/human/H)
+    if(H.mind && !player_is_antag(H.mind, only_offstation_roles = 1) && SSjobs.ShouldCreateRecords(H.mind.assigned_role))
```

Comment:
I wonder if the is antag check should be merged into `SSjobs.ShouldCreateRecords()`.